### PR TITLE
Revert change to rustc SwitchTargets type.

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

We have changed the visibility of one of the members of SwitchTargets to have access to their values during codegen. I re-wrote the code to use standard API so we don't need to keep using a forked version of the compiler. Issue #42.

The only changes left are the ones to enable rmc codegen and a a few options we enabled. Those should be handled once we have a new driver.

### Resolved issues:

None

### Testing:

* How is this change tested? Added new test + all older tests.

* Is this a refactor change? Yes

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: src/test/rmc/Match/match_bool.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// compile-flags: --crate-type lib
+// rmc-flags: --function match_bool
+
+#[rmc::proof]
+pub fn match_bool(arg: bool) {
```
Comment (tedinski) [inline src/test/rmc/Match/match_bool.rs:7]: I wouldn't use proof annotations on functions with arguments. This isn't really supported (yet!) and only sometimes sorta works.

**Review by tedinski — APPROVED** [review]

File: src/test/rmc/Match/match_bool.rs (hunk 1 above)
Comment (celinval) [reply src/test/rmc/Match/match_bool.rs:7]: I kind of prefer using the proof annotation than the `#[no_mangle]`. What are the downsides to it?

Comment (tedinski) [reply src/test/rmc/Match/match_bool.rs:7]: I'm just suggesting switching from an argument to `let arg = rmc::nondet();` at the start.

Comment (celinval) [reply src/test/rmc/Match/match_bool.rs:7]: I misunderstood you. Sorry about that. Sure, I can do that

**Pull request merged**